circuits/offchain/credentialAtomicQueryV3OffChain.circom

```diff
@@ -0,0 +1,269 @@
+        noAux <== issuerClaimNonRevMtpNoAux,
+        auxHi <== issuerClaimNonRevMtpAuxHi,
+        auxHv <== issuerClaimNonRevMtpAuxHv,
+        treeRoot <== issuerClaimNonRevRevTreeRoot
+    );
+
+    // verify issuer state for claim non-revocation proof
+    checkIdenStateMatchesRoots()(
+        1,
+        issuerClaimNonRevClaimsTreeRoot,
+        issuerClaimNonRevRevTreeRoot,
+        issuerClaimNonRevRootsTreeRoot,
+        issuerClaimNonRevState
+    );
+
+    component merklize = getClaimMerklizeRoot();
+    merklize.claim <== issuerClaim;
+
+    merklized <== merklize.flag;
+
+    // check path/in node exists in merkletree specified by jsonldRoot
+    SMTVerifier(claimLevels)(
+        enabled <== merklize.flag,  // if merklize flag 0 skip MTP verification
+        fnc <== claimPathNotExists, // inclusion
+        root <== merklize.out,
+        siblings <== claimPathMtp,
+        oldKey <== claimPathMtpAuxHi,
+        oldValue <== claimPathMtpAuxHv,
+        isOld0 <== claimPathMtpNoAux,
+        key <== claimPathKey,
+        value <== claimPathValue
+    );
+
+    // select value from claim by slot index (0-7)
+    signal slotValue <== getValueByIndex()(issuerClaim, slotIndex);
+
+    // select value for query verification,
+    // if claim is merklized merklizeFlag = `1|2`, take claimPathValue
+    // if not merklized merklizeFlag = `0`, take value from selected slot
+    signal fieldValue <== Mux1()(
+        [slotValue, claimPathValue],
+        merklize.flag
+    );
+
+    /////////////////////////////////////////////////////////////////
+
+    // verify query
+    signal querySatisfied <== Query(valueArraySize)(
+        in <== fieldValue,
+        value <== value,
+        operator <== operator
+    );
+
+    querySatisfied === 1;
+
+    /* ProfileID calculation */
+    userID <== SelectProfile()(userGenesisID, profileNonce);
+}
+
+template sigFlow(issuerLevels) {
+    signal input enabled;
+    signal input issuerAuthClaim[8];
+    signal input issuerAuthClaimsTreeRoot;
+    signal input issuerAuthRevTreeRoot;
+    signal input issuerAuthRootsTreeRoot;
+    signal input issuerAuthClaimMtp[issuerLevels];
+    signal input issuerAuthClaimNonRevMtp[issuerLevels];
+    signal input issuerAuthClaimNonRevMtpNoAux;
+    signal input issuerAuthClaimNonRevMtpAuxHi;
+    signal input issuerAuthClaimNonRevMtpAuxHv;
+    signal input issuerClaimNonRevRevTreeRoot;
+    signal input issuerClaim[8];
+    signal input issuerClaimSignatureR8x;
+    signal input issuerClaimSignatureR8y;
+    signal input issuerClaimSignatureS;
+    signal output issuerAuthState;
+
+    verifyCredentialSchema()(
+        enabled,
+        issuerAuthClaim,
+        80551937543569765027552589160822318028
+    );
+
+    signal tmpAuthState;
+    tmpAuthState <== getIdenState()(
+        issuerAuthClaimsTreeRoot,
+        issuerAuthRevTreeRoot,
+        issuerAuthRootsTreeRoot
+    );
+    issuerAuthState <-- enabled ==1 ? tmpAuthState : 0;
```

Comment:
Same problem with <--, change to similar code as proposed by Andriian.

Reply:
done